src/Controller/ComponentRegistry.php

```diff
@@ -150,22 +153,41 @@ protected function _create(object|string $class, string $alias, array $config):
             return $class;
         }
         if ($this->container?->has($class)) {
-            $constructor = (new ReflectionClass($class))->getConstructor();
-
-            if ($constructor !== null) {
-                $args = $this->reflectArguments($constructor, ['config' => $config]);
-
-                try {
-                    $this->container->extend($class)
-                        ->addArguments($args);
-                } catch (NotFoundException) {
-                    $this->container->add($class)
-                        ->addArguments($args);
+            // Try to resolve directly from container first
+            // This allows user-configured components to work without modification
+            try {
+                /** @var \Cake\Controller\Component $instance */
+                $instance = $this->container->get($class);
```

Comment:
Instead of a try catch could you use `container->has($class)` to see if the container already has the component defined?